libraries/apollo-compiler/src/test/kotlin/com/apollographql/apollo3/compiler/keyfields/KeyFieldsTest.kt

```diff
@@ -125,15 +115,9 @@ class KeyFieldsTest {
 
     val (operationDefinitions, schemaDefinitions) = definitions.partition { it is GQLExecutableDefinition }
 
-    try {
-      val schema = GQLDocument(schemaDefinitions, null).validateAsSchema().getOrThrow()
-      var operation = (operationDefinitions.single() as GQLOperationDefinition)
-      operation = addRequiredFields(operation, "always", schema, emptyMap())
-      checkKeyFields(operation, schema, emptyMap())
-      fail("An exception was expected")
-    } catch (e: SourceAwareException) {
-      check(e.message!!.contains("Field 'id' is not a valid key field for type 'Animal'"))
-      return
-    }
+    val issues = GQLDocument(schemaDefinitions, null).validateAsSchema().issues
+    check(issues.any { it.message.contains("Field 'id' is not a valid key field for type 'Animal'") })
+    check(issues.any { it.message.contains("Field 'version' is not a valid key field for type 'Node'") })
+    check(issues.any { it.message.contains("Field 'x' is not a valid key field for type 'Foo'") })
```

Comment:
Nit: I'd change the error to something like 

```
"No such field: 'Foo.x'"
```

I think there's no need to mention "key" since it's easy to see from the context and sourceLocation.